app/src/main/java/com/kickstarter/viewmodels/DiscoveryFragmentViewModel.kt

```diff
@@ -0,0 +1,583 @@
+            ) { _, params ->
+                params
+            }
+
+            val startOverWith = Observable.merge(
+                selectedParams,
+                selectedParams.compose(Transformers.takeWhen(refresh))
+            )
+
+            val paginator = builder<Project, DiscoverEnvelope, DiscoveryParams?>()
+                .nextPage(nextPage)
+                .distinctUntilChanged(true)
+                .startOverWith(startOverWith)
+                .envelopeToListOfData { it.projects() }
+                .loadWithParams {
+                    makeCallWithParams(
+                        it
+                    )
+                }
+                .clearWhenStartingOver(false)
+                .concater { xs, ys ->
+                    xs?.let { firstList ->
+                        ys?.let { secondList ->
+                            ListUtils.concatDistinct(firstList, secondList)
+                        }
+                    }
+                }
+                .build()
+
+            paginator.isFetching()
+                .compose(bindToLifecycle())
+                .subscribe(isFetchingProjects)
+
+            projectList
+                .compose(Transformers.ignoreValues())
+                .compose(bindToLifecycle())
+                .subscribe { isFetchingProjects.onNext(false) }
+
+            val activitySampleProjectClick = activitySampleProjectClick
+                .map<Pair<Project, RefTag>> {
+                    Pair.create(
+                        it,
+                        RefTag.activitySample()
+                    )
+                }
+
+            projectCardClicked
+                .compose(bindToLifecycle())
+                .subscribe {
+                    analyticEvents.trackProjectCardClicked(it, EventContextValues.ContextPageName.DISCOVER.contextName)
+                }
+
+            paramsFromActivity
+                .compose(Transformers.takePairWhen(projectCardClicked))
+                .compose(bindToLifecycle())
+                .subscribe {
+                    val refTag =
+                        RefTagUtils.projectAndRefTagFromParamsAndProject(it.first, it.second)
+                    val cookieRefTag = RefTagUtils.storedCookieRefTagForProject(
+                        it.second,
+                        cookieManager,
+                        sharedPreferences
+                    )
+                    val projectData = builder()
+                        .refTagFromIntent(refTag.second)
+                        .refTagFromCookie(cookieRefTag)
+                        .project(it.second)
+                        .build()
+                    analyticEvents.trackDiscoverProjectCtaClicked(it.first, projectData)
+                }
+
+            val projectCardClick = paramsFromActivity
+                .compose(Transformers.takePairWhen(projectCardClicked))
+                .map {
+                    RefTagUtils.projectAndRefTagFromParamsAndProject(it.first, it.second!!)
```

Comment:
Here as well, not sure if we should force this open. What do you think?

Reply:
Done☺️